### PR TITLE
Improve setup db script

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "some tweaks".

Proposed changes include:
* Move some code into tools/setup-mssql-database.sh to allow calling it separately, when needed
* Remove unused stuff from mssql schema
* Rewrite setup_riak script into erlang (because it's faaaster)
* `-dsaparam` optimization goes away :( (but we cache certs anyway, so...).
* But we improve starting time of ldap container, because we don't need to generate dh pair (we reuse mongooseim pair).
* Fix a bug with path calculation, allowing to run start_setup_db from any directory
* `docker rm` removes volumes now (finally)
* Make database ports configurable using env variables (we don't use the feature yet)
* Use `docker exec` to setup cassandra schemas (speedup)
* Remove "docker -i" attribute (to allow to run from non-interactive shell).

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7186.1](https://travis-ci.org/esl/MongooseIM/jobs/585650641) / Erlang 22.0 / small_tests / d5e2a6884142ad53c603e64d75b19f0844f8f7d9
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2459/7186/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2459/7186/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-16_19.38.06/lib.mongooseim.logs/run.2019-09-16_19.38.06/suite.log.html)

**Comment by mongoose-im** [comment]
[7187.1](https://travis-ci.org/esl/MongooseIM/jobs/585799410) / Erlang 22.0 / small_tests / bc4eeee4bff7e2be2a475892dcda4e16cf8238dc
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2459/7187/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2459/7187/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-16_21.57.29/lib.mongooseim.logs/run.2019-09-16_21.57.29/suite.log.html)

File: tools/travis-setup-ldap.sh [hunk 1]
```diff
@@ -20,42 +22,65 @@ echo "LDAP_ROOT_DIR=$LDAP_ROOT_DIR"
 
 mkdir -p "$LDAP_SCHEMAS_DIR" "$LDAP_DATA_DIR" "$LDAP_CONFIG_DIR" "$LDAP_CERT_DIR"
 
-cat > "$LDAP_SCHEMAS_DIR/init_entries.ldif" << EOL
-dn: ou=Users,dc=esl,dc=com
+function write_init_entries
+{
+cat > "$LDAP_SCHEMAS_DIR/init_entries$1.ldif" << EOL
```
Comment (NelsonVides) [inline tools/travis-setup-ldap.sh]: Indent man, it's a function! 😆

File: tools/setup_riak.escript [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    [wait_until_bucket_type_status(Type, ExpectedStatus, Node) || Node <- Nodes];
+wait_until_bucket_type_status(Type, ExpectedStatus, Node) ->
+    F = fun() ->
+                ActualStatus = rpc:call(Node, riak_core_bucket_type, status, [Type]),
+                ExpectedStatus =:= ActualStatus
+        end,
+    ok = wait_until(F, 30, 100).
+
+
+%% @doc Retry `Fun' until it returns `Retry' times, waiting `Delay'
```
Comment (NelsonVides) [inline tools/setup_riak.escript]: Should read:
> Retry `Fun` until it returns `true`, `Retry` times, waiting...

File: tools/travis-setup-ldap.sh (hunk 1 above)
Comment (NelsonVides) [reply tools/travis-setup-ldap.sh]: Ah, it's a `cat`, I see xD

Comment (arcusfelis) [reply tools/travis-setup-ldap.sh]: heredoc :)

**Pull request merged**